### PR TITLE
Show loading screen until API is ready

### LINKED ISSUE #519: Add loading screen to indicate the App is connecting to the network (API initialization)
- Use the loading screen we used to have and add a text to let the user know that it could take several seconds for the app to connect to the network. 
- After the API is ready show the Dashboard.
- When switching networks the loading screen should be displayed as well.
- In case of errors, allow the user to switch the network. (show a button to open the modal)

### PR DESCRIPTION
**Pull request by ishrak-imam**
### Description

Fixes https://github.com/litentry/litentry-app/issues/519

Restructure the navigation to show loading screen until the API is ready.

### CODE REVIEW

File: src/navigation/AppNavigator.tsx [hunk 1]
```diff
@@ -298,16 +342,11 @@ function AppNavigator() {
...
         }}>
-        {pushAuthorizationStatus && pushAuthorizationStatus === messaging.AuthorizationStatus.NOT_DETERMINED ? (
-          <AppStack.Screen name={routeKeys.permissionGrantingPromptScreen} component={PermissionGrantingPrompt} />
-        ) : undefined}
-        <AppStack.Screen name={routeKeys.drawerNavigatorScreen} component={DrawerNavigator} />
-        <AppStack.Screen name={routeKeys.addAccountScreen} component={AddAccountScreen} />
-        <AppStack.Screen name={routeKeys.balanceScreen} component={BalanceScreen} />
-        <AppStack.Screen name={routeKeys.identityGuideScreen} component={IdentityGuideScreen} />
-      </AppStack.Navigator>
+        {api ? <RootStack.Screen name={routeKeys.appStack} component={AppNavigator} /> : undefined}
```
Comment (ahoseinian) [inline src/navigation/AppNavigator.tsx]: shouldn't this be `api.status === 'ready'` or something? I think the api object get's filled immediately after app start

File: src/screen/ApiLoadingScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import {appStack, networkSelectionScreen} from 'src/navigation/routeKeys';
+
+export function ApiLoadingScreen({navigation}: {navigation: NavigationProp<any>}) {
+  const {currentNetwork} = useContext(NetworkContext);
+  const {api} = useContext(ChainApiContext);
+
+  // navigate to app in case of successfully connecting
+  const prevApi = React.useRef(!!api);
+  React.useEffect(() => {
+    if (api && !prevApi.current) {
```
Comment (ahoseinian) [inline src/screen/ApiLoadingScreen.tsx]: again here probably we need to check when api is ready

**Review by silva-fj — DISMISSED** [review]

File: src/screen/NetworkSelectionScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      closeOnOverlayTap={false}
+      panGestureEnabled={false}
+      tapGestureEnabled={false}
+      onClose={onClose}>
+      <View style={styles.networkModal}>
+        <NetworkSelectionList
+          items={availableNetworks}
+          selected={currentNetwork}
+          onSelect={(item) => {
+            select(item);
```
Comment (ahoseinian) [inline src/screen/NetworkSelectionScreen.tsx:53]: should we add 
```
navigation.navigate(apiLoadingScreen);
```
here? so the user sees we are loading the screen?

File: src/screen/ApiLoadingScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  }, [navigation, status]);
+
+  return (
+    <Layout style={styles.container}>
+      <ScreenNavigation
+        renderTitle={() => (
+          <TouchableOpacity onPress={() => navigation.navigate(networkSelectionScreen)}>
+            <Layout style={{alignItems: 'center'}}>
+              <Text category="s1">Litentry</Text>
+              {currentNetwork ? <NetworkItem item={currentNetwork} isConnected={false} /> : null}
```
Comment (ahoseinian) [inline src/screen/ApiLoadingScreen.tsx]:
```suggestion
              {currentNetwork ? <NetworkItem item={currentNetwork} isConnected={status === 'ready'} /> : null}
```

File: src/screen/ApiLoadingScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+            </Layout>
+          </TouchableOpacity>
+        )}
+      />
+      <View style={globalStyles.centeredContainer}>
+        <View style={styles.textContainer}>
+          <Icon style={[globalStyles.inlineIconDimension, {color: colorGreen}]} name="planet" pack="ionic" />
+          <Text style={styles.text}>Connecting to {currentNetwork?.name ?? ''}</Text>
```
Comment (ahoseinian) [inline src/screen/ApiLoadingScreen.tsx]: also probably not show connecting when it's ready

**Review by ahoseinian — DISMISSED** [review]

**Review by ahoseinian — APPROVED** [review]

**Pull request merged**